### PR TITLE
chore: update eslint@7.32.0 to 8.10.0

### PR DESCRIPTION
**Pull request by Trott**
Also update eslint-plugin-promise@5.2.0 to 6.0.0.

### CODE REVIEW

**Comment by XhmikosR** [comment]
@Trott might be a good chance to specify ecmaVersion or es**? Otherwise feel free to revert the .eslintignore change, I just happened to find this.

File: .eslintignore [hunk 1]
```diff
@@ -2,5 +2,3 @@
 static/legacy/
 external/
 build/
```
Comment (Trott) [inline .eslintignore]:
```suggestion
build/
# Top level await isn't supported till ESLint 8
locale/en/blog/release/v17.0.0.md
```

**Comment by Trott** [comment]
> @Trott might be a good chance to specify ecmaVersion or es**?

Fine with me, although maybe in a subsequent PR?

> Otherwise feel free to revert the .eslintignore change, I just happened to find this.

Yeah, reverted. Thanks.

**Comment by Trott** [comment]
@nodejs/website This is very straightforward but needs a review.

**Review by nschonni — APPROVED** [review]

**Pull request merged**